components/powersensor/collectors.go

```diff
@@ -0,0 +1,41 @@
+package powersensor
+
+import (
+	"context"
+
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"go.viam.com/rdk/data"
+)
+
+func assertPowerSensor(resource interface{}) (PowerSensor, error) {
+	ps, ok := resource.(PowerSensor)
+	if !ok {
+		return nil, data.InvalidInterfaceErr(API)
+	}
+	return ps, nil
+}
+
+type lowLevelCollector func(ctx context.Context, ps PowerSensor) (interface{}, error)
```

Comment:
[q] maybe a stupid question but where is this used?

Reply:
Only movementsensor also has this type though, the other components have a separate collector method for each function. Which would be preferred? @agreenb